### PR TITLE
bit stuff

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: content/4_Gold/PURS.mdx [hunk 1]
```diff
@@ -739,28 +802,40 @@ Covered in [Platinum](/plat/seg-ext).
...
+		
+		int arr_len;
+		cin >> arr_len;
+		long long inv_num = 0;
+		for (int i = 0; i < arr_len; i++) {
+			int x;
+			cin >> x;
+			/*
+			 * .order_of_key(x) gives the # of elements
+			 * in the tree that are greater than x.
```
Comment (ryanchou-dev) [inline content/4_Gold/PURS.mdx:832]: strictly less than right?

**Review by ryanchou-dev — APPROVED** [review]

File: content/4_Gold/PURS.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/4_Gold/PURS.mdx:832]: image
this is the OG comment

Comment (ryanchou-dev) [reply content/4_Gold/PURS.mdx:832]: image

Comment (ryanchou-dev) [reply content/4_Gold/PURS.mdx:832]: wait lemme reread this

Comment (ryanchou-dev) [reply content/4_Gold/PURS.mdx:832]: oh yeah you're right

Comment (SansPapyrus683) [reply content/4_Gold/PURS.mdx:832]: actually no i think you were, the og comment was just misleading as frick

Comment (ryanchou-dev) [reply content/4_Gold/PURS.mdx]: the problem is asking for something different though?

Comment (SansPapyrus683) [reply content/4_Gold/PURS.mdx]: we want the # of inversions
so for each element, we want the # of elements preceding it that are greater than said element

Comment (ryanchou-dev) [reply content/4_Gold/PURS.mdx]: thought maybe they passed in some cryptic stuff in the type definition to make it go the other way

File: content/4_Gold/PURS.mdx [hunk 2]
```diff
@@ -512,38 +512,81 @@ glance.
...
+		int size;
+		vector<T> bit;
+		vector<T> arr;
+	public:
+		BIT(int size) : size(size), bit(size + 1), arr(size) { }
 
-void upd(int i, ll v) {
-	for (; i<=n; i+=i&(-i)) bit[i]+=v;
-}
+		/** Sets the value of index ind at the actual array to vall. */
```
Comment (envyaims) [inline content/4_Gold/PURS.mdx:537]:
```suggestion
		/** Sets the value at index ind to val. */
```

File: content/4_Gold/PURS.mdx [hunk 3]
```diff
@@ -512,38 +512,81 @@ glance.
...
+using std::cout;
+using std::endl;
+using std::vector;
 
-int n, q;
-vector<ll> bit(MX), x(MX);
+/**
+ * Short for "binary indexed tree",
+ * this data structure supports point update and range sum
+ * queries like a segement tree.
```
Comment (envyaims) [inline content/4_Gold/PURS.mdx:526]:
```suggestion
 * queries like a segment tree.
```

File: content/4_Gold/PURS.mdx [hunk 4]
```diff
@@ -512,38 +512,81 @@ glance.
...
+		void add(int ind, int val) {
+			arr[ind] += val;
+			ind++;
+			for (; ind <= size; ind += ind & -ind) {
+				bit[ind] += val;
+			}
+		}
+
+		/** @return The sum of all values in [0, ind]. */
+		T pref_sum(int ind) {
```
Comment (envyaims) [inline content/4_Gold/PURS.mdx:552]: why not just call it `sum`

File: content/4_Gold/PURS.mdx [hunk 5]
```diff
@@ -512,38 +512,81 @@ glance.
...
-			cout << qry(b)-qry(a-1) << "\n";
+
+	for (int q = 0; q < query_num; q++) {
+		int type;
+		std::cin >> type;
+		if (type == 1) {
+			int ind;
+			int val;
+			std::cin >> ind >> val;
+			bit.set(ind - 1, val);
```
Comment (envyaims) [inline content/4_Gold/PURS.mdx:581]: either do `ind - 1` or `--ind`. Make up your mind

**Review by envyaims — CHANGES_REQUESTED** [review]
im ngl these changes feel a bit pointless but ok

File: content/4_Gold/PURS.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply content/4_Gold/PURS.mdx]: OOPS LMAO I THOUGHT IT WAS JUST order_of_key and not the i - stuff

File: content/4_Gold/PURS.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply content/4_Gold/PURS.mdx:552]: `sum`? `sum` of what?

**Review by dongliu0426 — APPROVED** [review]
neat

**Pull request merged**